lib/src/utils/keyify.ts

```diff
@@ -1,20 +1,20 @@
 export default function keyify(object: Record<string, unknown>, prefix: unknown[] = []): string[] {
-  return Object.keys(object).reduce<string[]>((result, element) => {
-    if (Array.isArray(object[element])) {
-      return result;
-    }
+  const result: string[] = [];
 
-    if (typeof object[element] === "object" && object[element] !== null) {
-      prefix.push(element);
-      const keys = keyify(object[element] as Record<string, unknown>, prefix);
-      prefix.pop();
+  for (const element of Object.keys(object)) {
```

Comment:
We could use `Object.entries` if we later use the value too.
See the code [here](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAawKYE8bHQCjgIwCtVoAuRAJRLgCcATAHgGcoaYwBzAGkXGTDgB3MAD4eABxqpgMAB7k+A4QG0AuogC8iNQEpyLNpzWIA3gChEiCAhaIpTEABso+1uw7GtagNxmLiYFpEHGswW2VUR1QAW1QwKB4ANwBDRxBUdThgRAB5IhIoADo4t1QmPHzoHR1Tf0ssYIBCAEEaGmT0QpgmVvbcFLTUatrLUcQGnCh0cVQsxAH0zQ0tACICYmgVxAAybfnUxcblxDAnRxrzMbHJaTlC8RAmAAscSJiSnV8r0dDbNHQmJoUBgsP0DqhEMlAVRrPRmG5ODxFEJRBIpDJZJ86lcbhj7nBxDgsdixvYnEUHs8cIUaf8mFjvgBfRCRJgQy7fKw2KABdipAAK6LkQNxdyinCgT0QAH5EKLZIVCHB2DgVoUVjUANSINVbcgrFZfTlk5z3R4vGRgAVC2SIbVvWLxBlXRnY12WV3+KRQEA0JAmqC+T2-OBRQqOOAcHD-UE4EwssDkeNW2LkADkKdQacZEkcSZOyVTiDTMGiMCziEZjOqQA).

I'm not sure about the performance though.